src/MessageList/MessageListBox.vala

```diff
@@ -30,26 +30,27 @@ public class Mail.MessageListBox : Gtk.ListBox {
         get_style_context ().add_class ("deck");
     }
 
-    public void set_conversation (Camel.FolderThreadNode node) {
+    public void set_conversation (Camel.FolderThreadNode? node) {
+        // Prevent the user from interacting with the message thread while it
+        // is being reloaded. can_reply will be set to true after loading the
+        // thread.
+        can_reply = false;
+
         get_children ().foreach ((child) => {
             child.destroy ();
         });
 
+        if (node == null)
+            return;
+
         var item = new MessageListItem (node.message);
         add (item);
         if (node.child != null) {
             go_down ((Camel.FolderThreadNode?) node.child);
         }
 
         var children = get_children ();
-        if (children.length () == 1) {
-            var child = get_row_at_index (0);
-            if (child is MessageListItem) {
-                var list_item = (MessageListItem) child;
-                list_item.expanded = true;
-                list_item.bind_property ("loaded", this, "can-reply", BindingFlags.SYNC_CREATE);
-            }
-        } else {
+        if (children.length () > 0) {
```

Comment:
I know that this was the case before but as we are using many time `children.length ()` it might be a good thing to cache this value as getting the length of an array can be an expensive operation